src/App.tsx

```diff
@@ -1,35 +1,472 @@
+            const hr = await decryptDataInKms(row.hr, kmsToken, keysId.masterSecretKeyUID, ACCESS_POLICY);
+            const all = { ...JSON.parse(marketing), ...JSON.parse(hr) };
+            return all;
+          })
+        );
+        setKmsClearData(clear);
+        setKmsEncryptedData(undefined);
+      }
+    }
+  };
+
+  // render
+
+  if (isLoading) {
+    return (
+      <Layout>
+        <Center style={{ minHeight: "100vh" }}>
+          <Spinner size="xl" />
+        </Center>
+      </Layout>
+    );
+  }
+
+  if (error) {
+    return (
+      <>
+        <Layout>
+          <Center style={{ minHeight: "100vh" }}>
+            {error && <>Error</>}
+            <Button onClick={() => loginWithRedirect()}>Login with Auth0</Button>
+          </Center>
+        </Layout>
+      </>
+    );
+  }
+
+  return (
+    <Layout>
+      <Flex flexDirection={"column"} gap="8">
+        {/* INTRO */}
+        <Heading as="h1" size="2xl">
+          Cosmian for Saas Applications
+        </Heading>
+        <Text fontSize="xl">
+          Test KMS + PKI basics actions combined with our attribute-based encryption scheme: CoverCrypt.
+          <br />
+          You can find documentation for all these actions in <Code>/src/utils/actions.ts</Code>.
+        </Text>
+        <Heading as="h2" size="lg">
+          Example: employees database
+        </Heading>
+        <Image
+          boxSize="100%"
+          maxWidth={900}
+          alignSelf={"center"}
+          objectFit="cover"
+          src={EmployeesDatabaseImage}
+          alt="Employees database schema"
+        />
+        <Image boxSize="100%" maxWidth={700} alignSelf={"center"} objectFit="cover" src={DatabaseSchema} alt="Database schema" />
+
+        <Box>
+          <Text fontSize="md">
+            Consider the following 2 policy axes, Department and Country which data are partitioned by the following attributes:
+          </Text>
+          <OrderedList>
+            <ListItem>Department: Marketing and HR</ListItem>
+            <ListItem>Country: France, Spain and Germany</ListItem>
+          </OrderedList>
+          <Text>
+            Each pair (Department, Country) constitutes one of the XX=XX data partitions.
+            <br />
+            With Cosmian attribute-based encryption scheme, the encryption key is public. Encrypting systems (Spark, data engineering
+            applications, ETLs, etc.) do not have to be secured and can directly hold the key, relaxing constraints on the infrastructure.
+            The public key can encrypt for any partition defined by the policy.
+            <br />
+            Decryption keys can decrypt a subset of the partitions defined by the policy.
+          </Text>
+        </Box>
+
+        {!isAuthenticated && <Button onClick={() => loginWithRedirect()}>Login with Auth0</Button>}
+
+        {isAuthenticated && (
+          <>
+            <Button onClick={handleLogout} style={{ position: "absolute", top: 20, right: 20 }}>
```

Comment:
Clicking on logout lead me to an error page - because redirection is made to `127.0.0.1:3000` and not `localhost:3000`